acquire/guis/gui.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+import logging
+import time
+from argparse import Namespace
+from threading import Thread
+
+log = logging.getLogger("gui")
+
+
+class GUI:
+    _instance = None
+    _progress = 0
+    _closed = False
+    _shard = 0
+
+    thread = None
+    folder = None
+    ready = False
+    auto_upload = None
+    upload_available = False
+
+    def __new__(cls, flavour: str = None, upload_available: bool = False):
+        # singleton+factory pattern
```

Comment:
Missing type hint and return type

Reply:
Are we going to use return types for constructors? Because when I grep "def __new__" I don't see it anywhere?